ethereum/api/src/test-support/java/org/hyperledger/besu/ethereum/api/tls/SelfSignedPfxStore.java

```diff
@@ -0,0 +1,241 @@
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateEncodingException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.time.Instant;
+import java.time.Period;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.apache.tuweni.net.tls.TLS;
+import org.bouncycastle.asn1.x500.RDN;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x500.style.BCStyle;
+import org.bouncycastle.asn1.x500.style.IETFUtils;
+import org.bouncycastle.asn1.x509.BasicConstraints;
+import org.bouncycastle.asn1.x509.Extension;
+import org.bouncycastle.asn1.x509.GeneralName;
+import org.bouncycastle.asn1.x509.GeneralNames;
+import org.bouncycastle.cert.X509v3CertificateBuilder;
+import org.bouncycastle.cert.jcajce.JcaX509CertificateConverter;
+import org.bouncycastle.cert.jcajce.JcaX509v3CertificateBuilder;
+import org.bouncycastle.jce.provider.BouncyCastleProvider;
+import org.bouncycastle.operator.ContentSigner;
+import org.bouncycastle.operator.jcajce.JcaContentSignerBuilder;
+
+/**
+ * Utility class to generate temporary self-signed certificates in PKCS12 format for testing
+ * purposes using BouncyCastle APIs.
+ *
+ * <p>Note: DO NOT USE IN PRODUCTION. The generated stores and files are marked to be deleted on JVM
+ * exit.
+ *
+ * <p>The generated certificate supports SAN extension for multiple DNS and IP addresses
+ */
+public final class SelfSignedPfxStore {
+  private static final char[] DEFAULT_PASSWORD = "changeit".toCharArray();
+  private static final String DEFAULT_DN = "CN=localhost";
+  private static final String DEFAULT_ALIAS = "test";
+  private static final BouncyCastleProvider BOUNCY_CASTLE_PROVIDER = new BouncyCastleProvider();
+  private static final boolean IS_CA = true;
+  private final Path parentPath;
+  private final String alias;
+  private final char[] password;
+  private final String distinguishedName;
+  private final List<String> sanHostNames;
+  private final List<String> sanIpAddresses;
+  private X509Certificate certificate;
+  private KeyPair keyPair;
+  private Path keyStore;
+  private Path trustStore;
+  private Path knownClientsFile;
+  private Path passwordFile;
+
+  private SelfSignedPfxStore(
+      final Path parentPath,
+      final String alias,
+      final char[] password,
+      final String distinguishedName,
+      final List<String> sanHostNames,
+      final List<String> sanIpAddresses) {
+    this.parentPath = parentPath;
```

Comment:
nit: parentPath is good for lots of things - but given this is multiple files, maybe its containingFolder?
What happens if this points to a file?

Reply:
Refactored and simplified.